src/hooks/useEncryptionStatus.ts

```diff
@@ -6,21 +6,41 @@ SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
 Please see LICENSE files in the repository root for full details.
 */
 
-import { MatrixClient, Room } from "matrix-js-sdk/src/matrix";
-import { useEffect, useState } from "react";
+import { CryptoEvent, MatrixClient, Room, RoomStateEvent } from "matrix-js-sdk/src/matrix";
+import { useEffect, useMemo, useState } from "react";
+import { throttle } from "lodash";
 
 import { E2EStatus, shieldStatusForRoom } from "../utils/ShieldUtils";
+import { useTypedEventEmitter } from "./useEventEmitter";
 
 export function useEncryptionStatus(client: MatrixClient, room: Room): E2EStatus | null {
     const [e2eStatus, setE2eStatus] = useState<E2EStatus | null>(null);
 
-    useEffect(() => {
-        if (client.getCrypto()) {
-            shieldStatusForRoom(client, room).then((e2eStatus) => {
-                setE2eStatus(e2eStatus);
-            });
-        }
-    }, [client, room]);
+    const updateEncryptionStatus = useMemo(
+        () =>
+            throttle(
+                () => {
+                    if (client.getCrypto()) {
+                        console.log("getting status");
```

Comment:
this is not a helpful console message

Reply:
ugh.  I thought I removed all my debugging messages, but I forgot that one.